blockchain/reactor.go

```diff
@@ -122,6 +122,27 @@ func (bcR *BlockchainReactor) RemovePeer(peer *p2p.Peer, reason interface{}) {
 	bcR.pool.RemovePeer(peer.Key)
 }
 
+// respondToPeer loads a block and sends it to the requesting peer, if we have it.
+// According to the Tendermint spec, if all nodes are honest, no node should
+// be requesting for a block that's non-existent.
+func (bcR *BlockchainReactor) respondToPeer(msg *bcBlockRequestMessage, src *p2p.Peer) (queued bool) {
+	// Got a request for a block. Respond with block if we have it.
+	block := bcR.store.LoadBlock(msg.Height)
+	if block != nil {
+		msg := &bcBlockResponseMessage{Block: block}
+		return src.TrySend(BlockchainChannel, struct{ BlockchainMessage }{msg})
+	}
+
+	// Otherwise peer is asking for things we don't have.
```

Comment:
Again, no need to duplicate the log message

```
Otherwise peer is asking for things we don't have.
Peer asking for a block we don't have
```

```
According to the Tendermint spec,
```

do we need to focus on this? maybe we can skip this part? i.e. just `if the peer asks ...`

Reply:
I've removed it entirely, thanks.